pkg/controllers/management/clusterprovisioner/provisioner.go

```diff
@@ -179,6 +194,22 @@ func (p *Provisioner) Remove(cluster *v3.Cluster) (runtime.Object, error) {
 }
 
 func (p *Provisioner) Updated(cluster *v3.Cluster) (runtime.Object, error) {
+	var err error
+
+	if shouldCleanHelmReleases(cluster) {
```

Comment:
Confirmed that this should work offline. It looks like the version is coming from kubernetes, and is set at compile time of the apiserver binary. Because of this, the version should only change if there is a new apiserver binary in place which is running the newer version of k8s. Since we only need the apiserver to be new to process these requests, this logic should be sufficient for our needs.